### DISCUSSION TITLE
Unable to move a window with no title bar.

### DISCUSSION THREAD

**Discussion by yvan-noume** [microsoft/microsoft-ui-xaml#9797] [Q&A]
Hello,
And thank you in advance for you answers.

I created a window with no title bar and tried to implement a way to move the window myself. I want it to be draggable anywhere.
At first I tried to use the Win32 api call: **SendMessage(IntPtr hWnd, int Msg, int wParam, int lParam)**, but it seems to be broken in WinUI 3 because it doesn't work but in winforms and wfp it does work. Here is an exerpt that works in wpf:
```
    public void move_window(object sender, MouseButtonEventArgs e)
    {
        ReleaseCapture();
        SendMessage(new WindowInteropHelper(this).Handle,
            WM_NCLBUTTONDOWN, HT_CAPTION, 0);
    }
```
Then I tried to move the window myself by using the position of the cursor. I did it successfully with the Win32 call: **(GetCursorPos(out POINTL lpPoint))**. It is relatively simple, I just had to take the new position of the cursor when the pointer moves and change the position of the window accordingly. It is not as smooth as the **SendMessage(...)** but it is good enough.

But I found out later that this approach doesn't work for touch screen since there is not mouse. So I tried multiple alternatives , all of them not yielding the same result as the position of the cursor. They are all very unstable. the most stable approach so far is using the GetCurrentPoint() from winUI 3 but it is not good enough. So please could you help me out?

I also saw that someone ask almost the same question here: https://github.com/microsoft/WindowsAppSDK/discussions/2962,  but still the person uses the position of the cursor.

**Comment by ghost1372** [comment]
you can do this easily, we can use TitleBar API (put a Grid on window):
```
<Window>

    <Grid x:Name="grd">
        
    </Grid>
</Window>


```

and:

```
public MainWindow()
{
    this.InitializeComponent();
    ExtendsContentIntoTitleBar = true;
    SetTitleBar(grd);

    if (AppWindow.Presenter is OverlappedPresenter presenter)
    {
        presenter.IsResizable = false;
        presenter.IsMaximizable = false;
        presenter.IsMinimizable = false;
        presenter.SetBorderAndTitleBar(false, false);
    }
}
```
![xx](https://github.com/microsoft/microsoft-ui-xaml/assets/9213496/190e228a-4579-40ec-8201-4cefc1a7daf8)


or you can use new Custom TitleBar API and set drag regions:
https://learn.microsoft.com/en-us/windows/apps/develop/title-bar

**Reply by yvan-noume** [reply]
Thank you. I haven't thought of this one. It surement helps. The only downside is that I have a lot a work defining all the interactive area of the window. And if those interactives areas change, I will have to do that again. I'll try to implement and see.
